src/main/java/com/cburch/logisim/fpga/gui/HdlColorRenderer.java

```diff
@@ -37,52 +40,89 @@ public HdlColorRenderer() {
-      }
+  public Component getTableCellRendererComponent(JTable table, Object infoObj, boolean isSelected,
+                                                 boolean hasFocus, int rowIdx, int columnIdx) {
+    // Some sanity checks to avoid NPE
+    final var value = (infoObj != null) ? (String) infoObj : "???";
+
+    // Set default cell colors
+    setBackground(javax.swing.UIManager.getColor("Table.background"));
+    setForeground(javax.swing.UIManager.getColor("Table.foreground"));
+
+    // First row is always a FPGS support status.
+    if (rowIdx == 0) {
+      return renderFpgaSupportRow(columnIdx, value);
+    }
+
+    // Render all the rest
+    if (value.equals(REQUIRED_FIELD_STRING)) {
+      // This field (mostly lables) is required to be set for FGPA support to work.
+      // But this is not an error if user do not really care FPGA at this moment, so let's color differently.
+      setCellColors(Color.ORANGE);
+      setHorizontalAlignment(JLabel.CENTER);
+      setText(S.get("FPGAHdlRequired"));
+    } else if ((value.length() == 7 || value.length() == 9) && value.indexOf('#') == 0) {
+      // Is this #RGB or #RGBA hex color value? If so, render the color
+      final var red = Integer.valueOf(value.substring(1, 3), 16);
+      final var green = Integer.valueOf(value.substring(3, 5), 16);
+      final var blue = Integer.valueOf(value.substring(5, 7), 16);
+      final var alpha = value.length() == 7 ? 255 : Integer.valueOf(value.substring(7, 9), 16);
+      final var bgColor = new Color(red, green, blue, alpha);
+      setCellColors(bgColor);
+      setText(value.toUpperCase());
       setHorizontalAlignment(JLabel.CENTER);
     } else {
-      String myInfo = (String) Info;
-      if (myInfo != null && myInfo.equals(REQUIRED_FIELD_STRING)) {
-        setBackground(Color.YELLOW);
-        setForeground(Color.BLUE);
-        setText("HDL Required");
-        setHorizontalAlignment(JLabel.CENTER);
-        setBorder(null);
-      } else if (myInfo != null
-          && myInfo.contains("#")
-          && myInfo.indexOf('#') == 0
-          && (myInfo.length() == 7 || myInfo.length() == 9)) {
-        int red, green, blue, alpha;
-        red = Integer.valueOf(myInfo.substring(1, 3), 16);
-        green = Integer.valueOf(myInfo.substring(3, 5), 16);
-        blue = Integer.valueOf(myInfo.substring(5, 7), 16);
-        alpha = myInfo.length() == 7 ? 255 : Integer.valueOf(myInfo.substring(7, 9), 16);
-        setBackground(new Color(red, green, blue, alpha));
-        setText("");
-        setBorder(null);
+      if (isSelected) {
+        setForeground(javax.swing.UIManager.getColor("Table.selectionForeground"));
+        setBackground(javax.swing.UIManager.getColor("Table.selectionBackground"));
+      }
+      setText(value);
+      setHorizontalAlignment(JLabel.LEFT);
+    }
+
+    setBorder(null);
+    return this;
+  }
+
+  /**
+   * First attribute table rows always shows component FPGA support status.
```

Comment:
Fix wording: "First attribute table row always shows the component's FPGA support status."

Reply:
Done.